ims/database/image.py

```diff
@@ -7,77 +8,81 @@
 # This class was written as per the Repository Model which allows us to change the DB in the future without changing
 # business code
 class ImageRepository:
-    def __init__(self):
-        self.connection = None
-
     # inserts the arguments into table
     # Commits if inserted successfully otherwise rollbacks if some issue occured and bubbles the exception
-    def insert(self, image_name, project_id, is_public=False):
-        try:
-            self.connection = DatabaseConnection()
-            img = Image()
-            img.name = image_name
-            img.project_id = project_id
-            img.is_public = is_public
-            self.connection.session.add(img)
-            self.connection.session.commit()
-        except SQLAlchemyError:
-            self.connection.session.rollback()
-            raise
-        finally:
-            self.connection.close()
+    def insert(self, image_name, project_id, is_public=False, id=None):
+        with DatabaseConnection() as connection:
+            try:
+                img = Image()
+                img.name = image_name
+                img.project_id = project_id
+                img.is_public = is_public
+                if id is not None:
```

Comment:
What is the use case for specifying 'id' here - isn't id your image table's primary key?  Same